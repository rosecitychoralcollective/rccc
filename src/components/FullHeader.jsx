import React, { useState } from 'react';
import {
  AppBar, IconButton, MenuItem, Menu, Popover, Tab, Tabs, Toolbar,
} from '@material-ui/core';
import MoreIcon from '@material-ui/icons/MoreVert';
import FacebookIcon from '@material-ui/icons/Facebook';
import EmailIcon from '@material-ui/icons/Email';
import PhoneIcon from '@material-ui/icons/Phone';
import availableStyles from '../resources/availableStyles';
import translateString from '../utils/StringHelper';

const t = translateString;

const FullHeader = ({ props }) => {
  const {
    setStyle,
    classes,
    tabs,
    tabIndex,
    changeTabs,
    handleFbClick,
    handleEmailClick,
    handlePhoneClick,
    handlePopperClose,
    popperAnchorEl,
  } = props;
  const [styleAnchorEl, setStyleAnchorEl] = useState(null);

  const menuItems = availableStyles.map((_, i) => `Style ${i + 1}`);

  const handleStyleMenuClick = (event) => {
    setStyleAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setStyleAnchorEl(null);
  };

  const handleMenuClick = (index) => {
    setStyle(index);
    handleClose();
  };

  const env = process.env.NODE_ENV;

  return (
    <AppBar
      className={classes.header}
    >
      <Toolbar>
        <div className={classes.tabs}>
          <Tabs
            value={tabIndex}
            onChange={changeTabs}
            aria-label="tab bar"
            variant="scrollable"
            scrollButtons="auto"
            classes={{ indicatorColor: classes.indicatorColor }}
          >
            {
              tabs.map((tab) => (
                <Tab key={tab.name} label={tab.name} />
              ))
            }
            {
              env === 'development' && (
                <div>
                  <IconButton
                    aria-controls="simple-menu"
                    aria-haspopup="true"
                    onClick={handleStyleMenuClick}
                    color="inherit"
                  >
                    <MoreIcon />
                  </IconButton>
                  <Menu
                    id="style-menu"
                    anchorEl={styleAnchorEl}
                    keepMounted
                    open={Boolean(styleAnchorEl)}
                    onClose={handleClose}
                  >
                    {menuItems.map((name, index) => (
                      <MenuItem
                        key={name}
                        onClick={() => handleMenuClick(index)}
                      >
                        {name}
                      </MenuItem>
                    ))}
                  </Menu>
                </div>
              )
            }
            <Popover
              open={Boolean(popperAnchorEl)}
              anchorEl={popperAnchorEl}
              onClose={handlePopperClose}
              anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'center',
              }}
              transformOrigin={{
                vertical: 'top',
                horizontal: 'center',
              }}
            >
              <div id="phone-number" className={classes.phonePopover}>{t('Phone-Number')}</div>
            </Popover>
          </Tabs>
        </div>
        <div className={classes.contactButtons}>
          <IconButton
            onClick={handleFbClick}
            color="inherit"
            id="facebook-button"
          >
            <FacebookIcon />
          </IconButton>
          <IconButton
            onClick={handleEmailClick}
            color="inherit"
            id="email-button"
          >
            <EmailIcon />
          </IconButton>
          <IconButton
            onClick={handlePhoneClick}
            color="inherit"
            id="phone-button"
          >
            <PhoneIcon />
          </IconButton>
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default FullHeader;
