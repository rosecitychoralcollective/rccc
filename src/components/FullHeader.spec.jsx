import React from 'react';
import { shallow } from 'enzyme';
import { IconButton, Tab, Tabs } from '@material-ui/core';
import FullHeader from './FullHeader';
import translateString from '../utils/StringHelper';

const t = translateString;

const tabs = [{ name: 'asdf', route: '/asdf' }, { name: 'fdsa', route: '/fdsa' }];
const tabIndex = 0;
const changeTabs = jest.fn();
const popperAnchorEl = null;
const classes = {};
const setStyle = jest.fn();
const handleFbClick = jest.fn();
const handleEmailClick = jest.fn();
const handlePhoneClick = jest.fn();
const props = {
  tabs,
  tabIndex,
  changeTabs,
  popperAnchorEl,
  classes,
  setStyle,
  handleFbClick,
  handleEmailClick,
  handlePhoneClick,
};
const OLD_ENV = process.env.NODE_ENV;

describe('Full header component', () => {
  let component;
  beforeEach(() => {
    jest.resetModules();
    process.env.NODE_ENV = OLD_ENV;
    component = shallow(<FullHeader props={props} />);
  });

  afterEach(() => {
    process.env.NODE_ENV = OLD_ENV;
    handleFbClick.mockReset();
    handleEmailClick.mockReset();
    handlePhoneClick.mockReset();
  });

  it('Renders without crashing', () => {
    expect(component.html()).toMatchSnapshot();
  });
  it('Does not render the style selector when env is not development', () => {
    expect(component.exists('#menu')).not.toBe(true);
    process.env.NODE_ENV = 'production';
    const devComponent = shallow(<FullHeader props={props} />);
    expect(devComponent.exists('#menu')).not.toBe(true);
  });
  it('Renders the style selector when env is development', () => {
    process.env.NODE_ENV = 'development';
    const devComponent = shallow(<FullHeader props={props} />);
    expect(devComponent.exists('#menu')).toBe(true);
  });
  it('Renders the same number of tabs as given in props', () => {
    expect(component.find(Tab)).toHaveLength(tabs.length);
  });
  it('Sets the name of a tab to the name given', () => {
    expect(component.find(Tab).some('[label="asdf"]')).toBe(true);
    expect(component.find(Tab).some('[label="fdsa"]')).toBe(true);
  });
  it('Calls changetabs when onChange state changes', () => {
    component.find(Tabs).simulate('change');
    expect(changeTabs).toHaveBeenCalled();
  });
  it('Renders three IconButton components', () => {
    expect(component.find(IconButton)).toHaveLength(3);
  });
  it('Clicks the facebook button and no other buttons', () => {
    component.find('#facebook-button').simulate('click');
    expect(handleFbClick).toHaveBeenCalled();
    expect(handleEmailClick).not.toHaveBeenCalled();
    expect(handlePhoneClick).not.toHaveBeenCalled();
  });
  it('Can click the Email button', () => {
    component.find('#email-button').simulate('click');
    expect(handleFbClick).not.toHaveBeenCalled();
    expect(handleEmailClick).toHaveBeenCalled();
    expect(handlePhoneClick).not.toHaveBeenCalled();
  });
  it('Can click the Phone button', () => {
    component.find('#phone-button').simulate('click');
    expect(handleFbClick).not.toHaveBeenCalled();
    expect(handleEmailClick).not.toHaveBeenCalled();
    expect(handlePhoneClick).toHaveBeenCalled();
  });
  it('Displays a phone number', () => {
    expect(component.find('#phone-number').text()).toBe(t('Phone-Number'));
  });
});
