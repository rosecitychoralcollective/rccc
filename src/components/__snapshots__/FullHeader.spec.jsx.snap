// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full header component Renders without crashing 1`] = `"<header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><div><div class=\\"MuiTabs-root\\"><div style=\\"width:99px;height:99px;position:absolute;top:-9999px;overflow:scroll\\" class=\\"MuiTabs-scrollable\\"></div><div class=\\"MuiTabs-scroller MuiTabs-scrollable\\" style=\\"overflow:hidden\\"><div aria-label=\\"tab bar\\" class=\\"MuiTabs-flexContainer\\" role=\\"tablist\\"><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"true\\"><span class=\\"MuiTab-wrapper\\">asdf</span><span class=\\"PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator\\"></span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" tabindex=\\"-1\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\"><span class=\\"MuiTab-wrapper\\">fdsa</span></button></div></div></div></div><div><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" id=\\"facebook-button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" id=\\"email-button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" id=\\"phone-button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z\\"></path></svg></span></button></div></div></header>"`;
